src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/MirroringTest/ConnectionOnMirroringTest.cs

```diff
@@ -28,17 +29,14 @@ public static void TestMultipleConnectionToMirroredServer()
                 builder.ConnectTimeout = 0;
 
                 TestWorker worker = new TestWorker(builder.ConnectionString);
-                Thread childThread = new Thread(() => worker.TestMultipleConnection());
-                childThread.Start();
+                Task childTask = Task.Factory.StartNew(() => worker.TestMultipleConnection(), TaskCreationOptions.LongRunning);
```

Comment:
I'm curious if we really need to be doing these as long running tasks or if this code is just so old that it predates the TPL and so we had no choice but use threads for everything. If it doesn't need to be long running, I'd say we drop that and use `Task.Run` or better yet make `TestMultipleConnection` be async.

Reply:
I think it's the age of the code. My starting point was to make sure that exceptions wouldn't crash testhost, so I was just thinking about how to make sure the second thread's exception propagated into the main thread. The TPL guarantees that, and I used LongRunning to guarantee that we would create a new thread rather than use something on the thread pool.